psu/psu.c

```diff
@@ -760,15 +765,46 @@ static const script_funct_t funcs[] = {
 
 int main(int argc, char *argv[])
 {
-	int res = -1;
+	int opt, res = -1;
 	script_t script;
 	hid_device *dev = NULL;
+	char *ptr;
 
-	if (argc < 2)
-		return usage(argv[0]);
+	for (;;) {
+		opt = getopt(argc, argv, "ht:");
+		if (opt == -1) {
+			break;
+		}
 
-	if (script_load(&script, argv[1]) != SCRIPT_OK)
-		return -1;
+		switch (opt) {
+			case 'h':
+				usage(argv[0]);
+				return EXIT_SUCCESS;
+
+			case 't':
+				usbWaitTime = strtoul(optarg, &ptr, 10);
+				if (optarg == ptr || usbWaitTime < 0) {
```

Comment:
`strtoul` used, so no way to receive negative `usbWaitTime`, check is always false (UB :D)

Reply:
Done, added check.